ecs/pod_creator.go

```diff
@@ -27,11 +30,134 @@ func NewBasicECSPodCreator(c cocoa.ECSClient, v cocoa.Vault) (*BasicECSPodCreato
 
 // CreatePod creates a new pod backed by AWS ECS.
 func (m *BasicECSPodCreator) CreatePod(ctx context.Context, opts ...*cocoa.ECSPodCreationOptions) (cocoa.ECSPod, error) {
-	return nil, errors.New("TODO: implement")
+	var out cocoa.ECSPod
+	var err error
+
+	mergedPodOptions := cocoa.MergeECSPodCreationOptions(opts...)
+
+	translatedIn, secrets, err := m.TranslateRegisterTaskDefinitionInput(ctx, mergedPodOptions)
+	if err != nil {
+		return nil, errors.Wrap(err, "translating the register task definition input to the correct type")
+	}
+
+	registerOut, err := m.client.RegisterTaskDefinition(ctx, translatedIn)
+	if err != nil {
+		return nil, errors.Wrap(err, "registering task definition")
+	}
+
+	taskDef := cocoa.ECSTaskDefinition{
+		ID:    registerOut.TaskDefinition.TaskDefinitionArn,
```

Comment:
If you want this to not panic, you have to check that `registerOut.TaskDefinition != nil` and that `registerOut.TaskDefinition.TaskDefinitionArn` is non-nil.